terrafrom validate do not support -check-variables option anymore

## Describe the Bug
From terrafrom v0.12+, terraform validate does not support `-check-variables` option anymore. See the [discussion in the thread](https://github.com/hashicorp/terraform/issues/21408#issuecomment-495259124). From v0.12+ it won't check variable by default. Is it possible to remove `-check-variables` from `module/terraform/Makefile`?

## Expected Behavior
Expected to see when using terraform version 0.13+, there's no error when running `make terraform/validate`

## Steps to Reproduce
Steps to reproduce the behavior:
1. Run `export TERRAFORM_VERSION = 0.15.4`
2. Clone `build-harness` (`curl -sSL -o .build-harness "https://git.io/build-harness"; echo .build-harness`)
2. Run `make terraform/install`
3. Run `terraform/get-modules`
4. Run `terraform/validte`
5. See error below:
```
╷
│ Error: Failed to parse command-line flags
│
│ flag provided but not defined: -check-variables
╵

For more help on using this command, run:
  terraform validate -help
make: *** [terraform/validate] Error 1
```

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: OSX v11.13.1
 - build-harness version: `0.58.0`
 - terraform version: `0.13+`

--- a/modules/github/Makefile.init
+++ b/modules/github/Makefile.init
@@ -27,12 +27,15 @@ $(GITHUB_TERRAFORM_TEMPLATES): $(addprefix $(BUILD_HARNESS_PATH)/templates/terra
 	cp $(BUILD_HARNESS_PATH)/templates/terraform/$@ $@
 	git ls-files --error-unmatch $@ 2>/dev/null || git add $@
 
+# Update context.tf if it is in place. Leave it in directory
+# if it did not previously exist, but do not add it to git
+# in case this is going to be a collection of submodules.
 github/init/context.tf:
 	curl -o context.tf -fsSL https://raw.githubusercontent.com/cloudposse/terraform-null-label/master/exports/context.tf
-	git ls-files --error-unmatch context.tf 2>/dev/null || git add context.tf
 	@if [[ -d examples/complete ]]; then \
 		cp -p context.tf examples/complete/context.tf ; \
 	fi
+	find . -xdev -mindepth 2 -name context.tf -exec cp -p context.tf {} \;
 
 .PHONY: $(GITHUB_TEMPLATES) $(GITHUB_TERRAFORM_TEMPLATES) github/init/context.tf
 

--- a/modules/terraform/Makefile
+++ b/modules/terraform/Makefile
@@ -1,6 +1,6 @@
 TMP ?= /tmp
 TERRAFORM ?= $(BUILD_HARNESS_PATH)/vendor/terraform
-TERRAFORM_VERSION ?= 0.12.26
+TERRAFORM_VERSION ?= 1.0.11
 TERRAFORM_URL ?= https://releases.hashicorp.com/terraform/$(TERRAFORM_VERSION)/terraform_$(TERRAFORM_VERSION)_$(OS)_$(BUILD_HARNESS_ARCH).zip
 
 ## Install terraform
@@ -16,7 +16,7 @@ terraform/install:
 
 ## Ensure all plugins can be fetched
 terraform/get-plugins:
-	@$(TERRAFORM) init -get-plugins -backend=false -input=false >/dev/null
+	@$echo terraform -get-plugins no longer supported
 
 ## Ensure all modules can be fetched
 terraform/get-modules:
@@ -24,11 +24,7 @@ terraform/get-modules:
 
 ## Basic terraform sanity check
 terraform/validate:
-ifeq ("12","$(word 2, $(subst ., ,$(TERRAFORM_VERSION)))")
-	@echo "Terraform 0.12 does not support validate without skipping variables"
-else
-	@$(TERRAFORM) validate -check-variables=false
-endif
+	@$(TERRAFORM) validate
 
 ## Lint check Terraform
 terraform/lint:

--- a/templates/Makefile.build-harness
+++ b/templates/Makefile.build-harness
@@ -72,10 +72,15 @@ init::
 .PHONY : clean
 ## Clean build-harness
 clean::
-	@[ "$(BUILD_HARNESS_PATH)" == '/' ] || \
-	[ "$(BUILD_HARNESS_PATH)" == '.' ] || \
-	[ "$(BUILD_HARNESS_PATH)" == '/$(BUILD_HARNESS_PROJECT)' ] || \
-	echo rm -rf $(BUILD_HARNESS_PATH)
+	if [ -d "$(BUILD_HARNESS_PATH)" ]; then \
+	  if [ -d build-harness ] && [ $$(stat -f %i "$(BUILD_HARNESS_PATH)") == $$(stat -f %i build-harness) ]; then \
+	    rm -rf build-harness; \
+	  else \
+	    echo Not removing build harness from "$(BUILD_HARNESS_PATH)" because it appears to be shared.; \
+	    echo If you are sure you want to delete it, run: ; \
+	    echo '   rm -rf "$(BUILD_HARNESS_PATH)"'; \
+	  fi; \
+	fi
 
 .PHONY: build-harness/shell builder build-harness/shell/pull builder/pull builder/build
 

--- a/templates/README.yaml
+++ b/templates/README.yaml
@@ -1,0 +1,100 @@
+#
+# This is the canonical configuration for the `README.md`
+# Run `make readme` to rebuild the `README.md`
+#
+
+# Name of this project
+name:
+
+# Logo for this project
+#logo: docs/logo.png
+
+# License of this project
+license: "APACHE2"
+
+# Copyrights
+copyrights:
+- name: "Cloud Posse, LLC"
+  url: "https://cloudposse.com"
+  year: "2021"
+
+# Canonical GitHub repo
+github_repo:
+
+# Badges to display
+badges:
+- name: "Latest Release"
+  image: "https://img.shields.io/github/release/cloudposse/terraform-example-module.svg"
+  url: "https://github.com/cloudposse/terraform-example-module/releases/latest"
+- name: "Slack Community"
+  image: "https://slack.cloudposse.com/badge.svg"
+  url: "https://slack.cloudposse.com"
+- name: "Discourse Forum"
+  image: "https://img.shields.io/discourse/https/ask.sweetops.com/posts.svg"
+  url: "https://ask.sweetops.com/"
+
+# List any related terraform modules that this module may be used with or that this module depends on.
+related:
+- name: "terraform-null-label"
+  description: "Terraform module designed to generate consistent names and tags for resources. Use terraform-null-label to implement a strict naming convention."
+  url: "https://github.com/cloudposse/terraform-null-label"
+
+# List any resources helpful for someone to get started. For example, link to the hashicorp documentation or AWS documentation.
+references:
+
+# Short description of this project
+description: |-
+  Short
+  description
+
+# Introduction to the project
+introduction: |-
+  This is an introduction.
+
+# How to use this module. Should be an easy example to copy and paste.
+usage: |-
+  For a complete example, see [examples/complete](examples/complete).
+
+  For automated tests of the complete example using [bats](https://github.com/bats-core/bats-core) and [Terratest](https://github.com/gruntwork-io/terratest)
+  (which tests and deploys the example on AWS), see [test](test).
+
+  ```hcl
+  # Create a standard label resource. See [null-label](https://github.com/cloudposse/terraform-null-label/#terraform-null-label--)
+  module "label" {
+    source  = "cloudposse/label/null"
+    # Cloud Posse recommends pinning every module to a specific version, though usually you want to use the current one
+    # version = "x.x.x"
+
+    namespace = "eg"
+    name      = "example"
+  }
+
+  module "example" {
+    source  = "cloudposse/*****/aws"
+    # Cloud Posse recommends pinning every module to a specific version
+    # version = "x.x.x"
+
+    example = "Hello world!"
+
+    context = module.label.this
+  }
+  ```
+
+# Example usage
+examples: |-
+  Here is an example of using this module:
+  - [`examples/complete`](https://github.com/cloudposse/terraform-example-module/) - complete example of using this module
+
+# How to get started quickly
+quickstart: |-
+  Here's how to get started...
+
+# Other files to include in this README from the project folder
+include:
+- "docs/targets.md"
+- "docs/terraform.md"
+
+# Contributors to this project
+contributors:
+- name: "Erik Osterman"
+  github: "osterman"

